scyllacloud/hostdialer_test.go

```diff
@@ -0,0 +1,194 @@
+package scyllacloud
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/gocql/gocql"
+)
+
+func TestHostSNIDialer_ServerNameIdentifiers(t *testing.T) {
+	connectionStateCh := make(chan tls.ConnectionState, 1)
+
+	server := httptest.NewUnstartedServer(nil)
+	server.TLS = &tls.Config{
+		VerifyConnection: func(state tls.ConnectionState) error {
+			connectionStateCh <- state
+			return nil
+		},
+	}
+	server.StartTLS()
+	defer server.Close()
+
+	clientCert, clientKey, err := generateClientCert(server)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	certPem, err := encodeCertificates(clientCert)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	privateKeyPem, err := encodePrivateKey(clientKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	caCert, err := encodeCertificates(server.Certificate())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dialer := &gocql.ScyllaShardAwareDialer{Dialer: net.Dialer{}}
+	connConfig := &ConnectionConfig{
+		Datacenters: map[string]*Datacenter{
+			"us-east-1": {
+				CertificateAuthorityData: caCert,
+				Server:                   server.Listener.Addr().String(),
+				NodeDomain:               "node.domain.com",
+				// Test TLS server serving cert is signed for example.com domain,
```

Comment:
why can't we use matching IP if it isn't resolved?

Reply:
replaced with proper certificate validation